### ISSUE TITLE
feature request: link to PNG

### ISSUE THREAD

**Issue by andyczerwonka** [excalidraw/excalidraw#2895]
I'd be great if we were able to link to an image. I find myself wanting to do this all the time.

**Comment by dwelle** [comment]
You mean instead of exporting a shareable link to the excalidraw.com scene, you want a to export a link to the PNG file?

**Comment by andyczerwonka** [comment]
Exactly. I use the tool to draw an image, then I almost always want to embed an image somewhere.

**Comment by h7y** [comment]
How will the implementation look like? Maybe, have the option to generate a PNG link in context menu -> upload it to something like imgur and copy the image link to clipboard?

**Comment by lipis** [comment]
Using imgur or similar service would be OK.. but I don't think that we want to host/serve all those images. Keep in mind that at the moment we are storing encrypted data so we can't see the actual scene, while with images it's a different story.

**Comment by dincozdemir** [comment]
How about a renderer that converts a url link into an image?

**Comment by chipbite** [comment]
Perhaps an option to solve this need would be for excalidraw to be able to work with imagefiles (png/svg) with the drawing source embedded. I opened a ticket for this, #4891.

**Comment by devnoname120** [comment]
@dwelle Any chance this could be implemented soon? Being able to include an up-to-date image of diagrams would be tremendously helpful to offset the scarce number of integrations that Excalidraw+ has (for example no Jira and no Dropbox Paper support).

Excalidraw+ already has an endpoint that produces a JPEG preview of a given Excalidraw+ file. For example the preview of [this document](https://link.excalidraw.com/l/4plW8O1oVGb/88OffGkHoNk) can be found at [this address](https://firebasestorage.googleapis.com/v0/b/quickstart-1595168317408.appspot.com/o/previews%2F4plW8O1oVGb%2FVDmHQkBCQMIyy1RAMohN4mRZpmly9yahMxupsNXq?alt=media). The downside however is that the resolution is only 500 × 366 pixels which gives a blurry output…

I can use the following snippet to do something close to what I need:
```markdown
[![Excalidraw — feat. req — Direct link to PNG](https://firebasestorage.googleapis.com/v0/b/quickstart-1595168317408.appspot.com/o/previews%2F4plW8O1oVGb%2FVDmHQkBCQMIyy1RAMohN4mRZpmly9yahMxupsNXq?alt=media)](https://link.excalidraw.com/l/4plW8O1oVGb/88OffGkHoNk)
```

And here is the result:

[![Excalidraw — feat. req — Direct link to PNG](https://firebasestorage.googleapis.com/v0/b/quickstart-1595168317408.appspot.com/o/previews%2F4plW8O1oVGb%2FVDmHQkBCQMIyy1RAMohN4mRZpmly9yahMxupsNXq?alt=media)](https://link.excalidraw.com/l/4plW8O1oVGb/88OffGkHoNk)

**Comment by luigi-rosso** [comment]
Another +1 on this!

I find myself diagraming the logic of complex functions and dropping a link in an IDE will show the image on hover. Would be great if I wouldn't need to host the image myself. I can kind of do it by extracting the image url from the tags in the embed page, but it would be much nicer if this were natively supported.
![Image](https://github.com/user-attachments/assets/3c9f1383-39c3-48e9-b0f3-155272fac8e0)
